src/tools/rbd_mirror/instance_watcher/Types.h

```diff
@@ -39,8 +39,110 @@ WRITE_CLASS_ENCODER(PeerImageId);
 
 typedef std::set<PeerImageId> PeerImageIds;
 
+enum NotifyOp {
+  NOTIFY_OP_IMAGE_ACQUIRE  = 0,
+  NOTIFY_OP_IMAGE_RELEASE  = 1,
+};
+
+struct ImageAcquirePayload {
+  static const NotifyOp NOTIFY_OP = NOTIFY_OP_IMAGE_ACQUIRE;
+
+  uint64_t request_id;
+  std::string global_image_id;
+  PeerImageIds peers;
```

Comment:
Sorry, by peer (in the `PoolWatcher` context) I meant `ImagePeer`.

I am still not sure I see how we can nicely eliminate `ImagePeer` from the acquire/release notifications.

Let's we have two peers (remote mirroring pools). An image (primary) is created in one of the pools. The `PoolWatcher` sends update to the `Replayer`, and the `Replayer` sends image_acquire notification. At this moment the `Replayer` possesses information about the image peer (remote image) and can provide it within `image_acquire` notification. If it is not sent to the `InstanceReplayer`, it will need to figure this out itself before calling `ImageReplayer::add_remote_image`, i.e. scan all available peers for an image with this global id. Do you want me to do this way?